casper/src/main/resources/PoS.rhox

```diff
@@ -63,626 +66,646 @@ in {
-              for (@state <<- stateCh) {
-                returnCh!(state.get("activeValidators"))
-              }
-            } |
-
-            contract PoS (@"getActiveValidatorVaults", returnCh) = {
-              for(@vaults <<- perActiveValidatorPosVaultsCh) {
-                returnCh!(vaults)
-              }
-            } |
-
-            /**
-             * Returns a Map[PublidKey, Int] containing the rewards accumulated for each validator.
-             * The returned map contains only the committed rewards after the last closeBlock
-             */
-            contract PoS (@"getRewards", returnCh) = {
-              for (@state <<- stateCh) {
-                returnCh!(state.get("committedRewards"))
-              }
-            } |
-
-            contract PoS (@"bond", @deployerId, @amount, returnCh) = {
-              new userCh, depositCh, processCh in {
-                runMVar!(*stateCh, *processCh, *returnCh) |
-                getUser!(deployerId, *userCh) |
-                for(@state, resultCh <- processCh;
-                    @userPk <- userCh) {
-                  if (state.get("allBonds").contains(userPk)) {
-                    resultCh!(state, (false, "Public key is already bonded."))
-                  } else if (amount < $$minimumBond$$) {
-                    resultCh!(state, (false, "Bond is less than minimum!"))
-                  } else if (amount > $$maximumBond$$) {
-                    resultCh!(state, (false, "Bond is greater than maximum!"))
-                  } else {
-                    deposit!(deployerId, amount, state.get("activeValidators"), *depositCh) |
-                    for(@depositResult <- depositCh) {
-                      match depositResult {
-                        (true, _) => {
-                          resultCh!(
-                            state.set(
-                              "allBonds",
-                              state.get("allBonds").set(userPk,amount)),
-                            depositResult
-                          )
-                        }
-                        (false, errorMsg) => {
-                          resultCh!(state, (false, "Bond deposit failed: " ++ errorMsg))
-                        }
-                      }
-                    }
-                  }
+      // Creates the auth key and Rev vault associated with the PoS contract.
+      new posPkCh in {
+        @RevVault!("deployerAuthKey", *posId, *posAuthKeyCh) |
+        // Not required... but can be restored after real wallet public keys have been created.
+        //getCurrentUserAddress!(*posId, *posRevAddressCh) |
+        getUser!(*posId, *posPkCh) |
+        for (@posPk <- posPkCh) {
+          revAddressOps!("fromPublicKey", posPk, *posRevAddressCh) |
+          // Creates Coop multisig vault for slashed funds.
+          // (false, errorMsg) xor (true, (multiSigVault, revAddress, revVault)) are returned on coopWalletCh.
```

Comment:
Please say "either ... or".  It took me way too long to figure out how you intended to xor tuples.